contracts/src/IBoundlessMarket.sol

```diff
@@ -36,9 +36,9 @@ interface IBoundlessMarket {
     event RequestFulfilled(RequestId indexed requestId);
 
     /// @notice Event logged when a proof is delivered that satisfies the request's requirements.
-    /// @dev It is possible for this event to be logged multiple times for a single request. This
-    /// is usually logged as part of order fulfillment, however it can also be logged by a prover
-    /// sending the proof without payment.
+    /// @dev It is possible for this event to be logged multiple times for a single request. The
+    /// first event logged will always coincide with the `RequestFulfilled` event, independently of
+    /// the payment outcome.
```

Comment:
I would change the slightly.
```suggestion
    /// first event logged will always coincide with the `RequestFulfilled` event and the fulfilled flag on the request being set.
```
And added a comment to the `RequestFulfilled` event that reads.
```solidity
/// @dev It is possible to a request to be fulfilled, and this event to be emitted, without the payment being transferred. Payment ma still be transferred later if the conditions are met.
```